src/main/groovy/jenkins/automation/utils/GhUtils.groovy

```diff
@@ -12,51 +14,87 @@ class GhUtils {
-                            String ghAuthId,
-                            Boolean ghPermitAll = false,
-                            Boolean ghPrHooks = true,
-                            String ghPrCron = '',
-                            String ghPrOrgsList = 'jenkins'
-    ) {
+    static void ghPrWatcher(context, Map ghPrConfig) {
+        def defaults = ghPrConfigDefaults
         context.with {
             scm {
                 git {
                     remote {
-                        github(ghProject, 'https', ghHostname)
+                        github(
+                            ghPrConfig.get('ghProject'),
+                            'https',
+                            ghPrConfig.get('ghHostname')
+                        )
                         refspec('+refs/pull/*:refs/remotes/origin/pr/*')
                     }
                     branch('${sha1}')
                 }
             }
             configure { node ->
                 node / 'triggers' / 'org.jenkinsci.plugins.ghprb.GhprbTrigger'( plugin: 'ghprb') {
-                    if (ghPrCron) {
-                        spec(ghPrCron)
-                        cron(ghPrCron)
-                    }
-                    else {
-                        spec()
-                        cron()
-                    }
+                    // Avoid 'null' in these blocks
+                    spec(ghPrConfig['ghPrCron'] ?: '')
+                    cron(ghPrConfig['ghPrCron'] ?: '')
                     allowMembersOfWhitelistedOrgsAsAdmin(true)
-                    orgslist(ghPrOrgsList)
-                    permitAll(ghPermitAll)
-                    useGitHubHooks(ghPrHooks)
-                    gitHubAuthId(ghAuthId)
+                    orgslist(ghPrConfig.get('ghPrOrgsList', defaults.ghPrOrgsList))
+                    permitAll(ghPrConfig.get('ghPermitAll', defaults.ghPermitAll))
+                    useGitHubHooks(ghPrConfig.get('ghPrHooks', defaults.ghPrHooks))
+                    gitHubAuthId(ghPrConfig.get('ghAuthId'))
+                    extensions {
+                        'org.jenkinsci.plugins.ghprb.extensions.status.GhprbSimpleStatus' {
+                            commitStatusContext(
+                                ghPrConfig.get(
+                                    'ghPrStatusContext',
+                                    defaults.ghPrStatusContext
+                                )
+                            )
+                            triggeredStatus()
+                            startedStatus()
+                            statusUrl()
+                            addTestResults(true)
+                            def ghPrMessage = ghPrConfig.get(
+                                'ghPrResultMessage',
+                                defaults.ghPrResultMessage
+                            )
+                            def confCompletedStatus = completedStatus()
+                            ghPrMessage.each{ buildStatus, ghMessage ->
```

Comment:
I considered moving this to its own function, but I couldn't really see this getting reused.

Reply:
I also debated not putting these options in a map since there's just three of them, this just _looked_ cleaner to me. Open to suggestions.